ldap/servers/slapd/back-ldbm/dblayer.c

```diff
@@ -1414,6 +1421,14 @@ dblayer_pop_pvt_txn(void)
     return;
 }
 
+void
+dblayer_destroy_txn_stack(void)
+{
+    /* Cleanup for the main thread to avoid false/positive leaks from libasan */
+    void *txn_stack = PR_GetThreadPrivate(thread_private_txn_stack);
```

Comment:
Unused variable, but why "get" it if you just "set" it to NULL on the next line?

Reply:
Indeed, something looks broken.
either the slapi_ch_free(&txn_stack); line is missing, or the  PR_GetThreadPrivate should be deleted.  (depending whether PR_SetThreadPrivate calls the cleanup callback or not)  I will double check
|

Reply:
Clearly the dblayer_cleanup_txn_stack is called when using PR_SetThreadPrivate ==> No need to get the data to release it
```
    #1 0x7f00d94601eb in slapi_ch_free /home/progier/sb/r67020/tst/source/389-ds-base/ldap/servers/slapd/ch_malloc.c:286
    #2 0x7f00d22a6876 in dblayer_cleanup_txn_stack /home/progier/sb/r67020/tst/source/389-ds-base/ldap/servers/slapd/back-ldbm/dblayer.c:1373
    #3 0x7f00d8b04920 in PR_SetThreadPrivate (/lib64/libnspr4.so+0x13920) (BuildId: e6dfe24882ade9ec65f7f1591a3e9e249e5a5780)
    #4 0x7f00d22aac9f in dblayer_destroy_txn_stack /home/progier/sb/r67020/tst/source/389-ds-base/ldap/servers/slapd/back-ldbm/dblayer.c:1429
    #5 0x7f00d22aad5d in dblayer_close /home/progier/sb/r67020/tst/source/389-ds-base/ldap/servers/slapd/back-ldbm/dblayer.c:500
```